homeassistant/components/fan/wemo.py

```diff
@@ -0,0 +1,322 @@
+        is no longer available. Use polling to detect if a device is available.
+        """
+        return True
+
+    @property
+    def unique_id(self):
+        """Return the ID of this WeMo humidifier."""
+        return self._serialnumber
+
+    @property
+    def name(self):
+        """Return the name of the humidifier if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on. Standby is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return true if switch is available."""
+        return self._available
+
+    @property
+    def icon(self):
+        """Return the icon of device based on its type."""
+        return 'mdi:water-percent'
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_CURRENT_HUMIDITY: self._current_humidity,
+            ATTR_TARGET_HUMIDITY: self._target_humidity,
+            ATTR_FAN_MODE: self._fan_mode,
+            ATTR_WATER_LEVEL: self._water_level,
+            ATTR_FILTER_LIFE: self._filter_life,
+            ATTR_FILTER_EXPIRED: self._filter_expired
+        }
+
+    @property
+    def speed(self) -> str:
+        """Return the current speed."""
+        return WEMO_FAN_SPEED_TO_HASS.get(self._fan_mode)
+
+    @property
+    def speed_list(self: FanEntity) -> list:
+        """Get the list of available speeds."""
+        return SUPPORTED_SPEEDS
+
+    @property
+    def supported_features(self: FanEntity) -> int:
+        """Flag supported features."""
+        return SUPPORTED_FEATURES
+
+    async def async_added_to_hass(self):
+        """Wemo humidifier added to HASS."""
+        # Define inside async context so we know our event loop
+        self._update_lock = asyncio.Lock()
+
+        registry = self.hass.components.wemo.SUBSCRIPTION_REGISTRY
+        await self.hass.async_add_job(registry.register, self.wemo)
+        registry.on(self.wemo, None, self._subscription_callback)
+
+    async def async_update(self):
+        """Update WeMo state.
+
+        Wemo has an aggressive retry logic that sometimes can take over a
+        minute to return. If we don't get a state after 5 seconds, assume the
+        Wemo humidifier is unreachable. If update goes through, it will be made
+        available again.
+        """
+        # If an update is in progress, we don't do anything
+        if self._update_lock.locked():
+            return
+
+        try:
+            with async_timeout.timeout(5):
+                await asyncio.shield(self._async_locked_update(True))
+        except asyncio.TimeoutError:
+            _LOGGER.warning('Lost connection to %s', self.name)
+            self._available = False
+
+    async def _async_locked_update(self, force_update):
+        """Try updating within an async lock."""
+        async with self._update_lock:
```

Comment:
I was using the wemo switch platform as a template, and they used this paradigm there. I'm assuming since it's an async subscription, that they're trying to avoid multiple concurrent updates for the same device from conflicting with each other.